pybamm/util.py

```diff
@@ -227,95 +226,48 @@ def __eq__(self, other):
 
 def load_function(filename):
     """
-    Load a python function from a file "function_name.py" called "function_name".
-    The filename might either be an absolute path, in which case that specific file will
-    be used, or the file will be searched for relative to PyBaMM root.
+    Load a python function from an absolute or relative path using `importlib`.
+    Example - pybamm.load_function("pybamm/input/example.py")
 
     Arguments
     ---------
     filename : str
-        The name of the file containing the function of the same name.
+        The path of the file containing the function.
 
     Returns
     -------
     function
         The python function loaded from the file.
     """
-
-    if not filename.endswith(".py"):
-        raise ValueError("Expected filename.py, but got {}".format(filename))
+    if filename.endswith(".py"):
+        filename = filename.replace(".py", "")
 
     if "lead-acid" in filename:
         warnings.simplefilter("always", DeprecationWarning)
         warnings.warn(
             "lead-acid is deprecated, use lead_acid instead", DeprecationWarning
         )
         filename = filename.replace("lead-acid", "lead_acid")
+
     if "lithium-ion" in filename:
         warnings.simplefilter("always", DeprecationWarning)
         warnings.warn(
             "lithium-ion is deprecated, use lithium_ion instead", DeprecationWarning
         )
         filename = filename.replace("lithium-ion", "lithium_ion")
 
-    # If it's an absolute path, find that exact file
-    if os.path.isabs(filename):
-        if not os.path.isfile(filename):
-            raise ValueError(
-                "{} is an absolute path, but the file is not found".format(filename)
-            )
-
-        valid_filename = filename
+    _, tail = os.path.split(filename)
 
-    # Else, search in the whole PyBaMM directory for matches
+    if "PyBaMM" in filename:
+        root_path = filename[filename.rfind("PyBaMM") :][7:]
```

Comment:
Maybe worth adding a comment explaining what this does and why?
Is it because you want to strip the full path to the module from `../PyBaMM/pybamm/input/example.py` to `pybamm/input/example.py` ?

Reply:
Yes, I'll add a comment

Reply:
So that works if `pybamm` is imported from a directory named `PyBaMM` (which is indeed default when cloning from github). But what happens if `pybamm` is imported from its installed location (e.g. `../site-packages/pybamm`)? This would be the case when installing PyBaMM with pip instead of importing it directly from the sources